### ISSUE TITLE
Missing locks for reconfiguring loggers

### ISSUE THREAD

**Issue by rnapier** [Sirupsen/logrus#241]
The `std` logger has locks available for reconfiguring it via `SetOutput`, `SetLevel`, `SetFormatter`, etc. But if you create a new logger, the only access is via the raw fields, which creates race conditions (and crashes in my case) if you try to modify them while logrus is logging.

Ideally, all of the `Set...` and `Get...` functions should be moved onto Logger methods, and the top-level functions should just forward to them for `std`. For safety, the struct fields should be unexported, but this would be a nontrivial change to the interface, and make it a little harder to initially configure loggers (unless some other "LoggerConfig" were provided). The current scheme, however, is very dangerous.

**Comment by Sirupsen** [comment]
You're completely right. I'm worried about the amount we'll break the standard interface, but on a path to 1.0 this is better done early rather than late.

**Comment by Astone-Chou** [comment]
"Ideally, all of the Set... and Get... functions should be moved onto Logger methods"

Really! Really! Really! Need to got the lock, plz.

**Comment by dmathieu** [comment]
We'll get to this issue at some point. In the mean time, feel free to open a PR if you want this to be fixed faster.

**Comment by dgsb** [comment]
has been fixed in 0c03a05a
